ratel-codegen/src/statement.rs

```diff
@@ -309,6 +310,72 @@ impl<'ast, G: Generator> ToCode<G> for SwitchCase<'ast> {
     }
 }
 
+impl<'ast, G: Generator> ToCode<G> for ImportDeclaration<'ast> {
+    #[inline]
+    fn to_code(&self, gen: &mut G) {
+        let mut tail = false;
+        gen.write_bytes(b"import ");
+        if self.specifiers.is_empty() {
+            gen.write_byte(b'\'');
+            gen.write_bytes(self.source.as_bytes());
+            gen.write_byte(b'\'');
+            return
+        }
+
+        for spec in self.specifiers {
+            match spec.item {
+                ForImportSpecifier::ImportDefaultSpecifier(import) => {
+                    if tail {
+                        gen.write_bytes(b", ");
+                    }
+                    gen.write(&import.local);
+                    tail = true;
+                },
+                ForImportSpecifier::ImportNamespaceSpecifier(import) => {
+                    if tail {
+                        gen.write_bytes(b", ");
+                    }
+                    gen.write_bytes(b"* as ");
+                    gen.write(&import.local);
+                    tail = true;
+                },
+                _ => {},
+            }
+        }
+        let mut need_close = false;
+        for spec in self.specifiers {
```

Comment:
Ditto about this loop, though here it's a bit more complicated due to the opening and closing brace. I'll write some suggestions for the code.